{
    "dataType": "CVE_RECORD",
    "cveMetadata": {
        "cveId": "CVE-2024-4032",
        "assignerOrgId": "28c92f92-d60d-412d-b760-e73465c3df22",
        "state": "PUBLISHED",
        "assignerShortName": "PSF",
        "dateReserved": "2024-04-22T17:15:47.895Z",
        "datePublished": "2024-06-17T15:05:58.827Z",
        "dateUpdated": "2024-07-05T18:17:37.383Z"
    },
    "containers": {
        "cna": {
            "affected": [
                {
                    "defaultStatus": "unaffected",
                    "product": "CPython",
                    "repo": "https://github.com/python/cpython",
                    "vendor": "Python Software Foundation",
                    "versions": [
                        {
                            "lessThan": "3.12.4",
                            "status": "affected",
                            "version": "0",
                            "versionType": "python"
                        },
                        {
                            "lessThan": "3.13.0a6",
                            "status": "affected",
                            "version": "3.13.0a1",
                            "versionType": "python"
                        }
                    ]
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "supportingMedia": [
                        {
                            "base64": false,
                            "type": "text/html",
                            "value": "<p><span style=\"background-color: transparent;\">The “ipaddress” module contained incorrect information about whether certain IPv4 and IPv6 addresses were designated as “globally reachable” or “private”. This affected the is_private and is_global properties of the ipaddress.IPv4Address, ipaddress.IPv4Network, ipaddress.IPv6Address, and ipaddress.IPv6Network classes, where values wouldn’t be returned in accordance with the latest information from the IANA Special-Purpose Address Registries.</span></p><p><span style=\"background-color: transparent;\">CPython 3.12.4 and 3.13.0a6 contain updated information from these registries and thus have the intended behavior.</span></p>"
                        }
                    ],
                    "value": "The “ipaddress” module contained incorrect information about whether certain IPv4 and IPv6 addresses were designated as “globally reachable” or “private”. This affected the is_private and is_global properties of the ipaddress.IPv4Address, ipaddress.IPv4Network, ipaddress.IPv6Address, and ipaddress.IPv6Network classes, where values wouldn’t be returned in accordance with the latest information from the IANA Special-Purpose Address Registries.\n\nCPython 3.12.4 and 3.13.0a6 contain updated information from these registries and thus have the intended behavior."
                }
            ],
            "providerMetadata": {
                "orgId": "28c92f92-d60d-412d-b760-e73465c3df22",
                "shortName": "PSF",
                "dateUpdated": "2024-06-17T15:11:36.402Z"
            },
            "references": [
                {
                    "tags": [
                        "issue-tracking"
                    ],
                    "url": "https://github.com/python/cpython/issues/113171"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/pull/113179"
                },
                {
                    "url": "https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml"
                },
                {
                    "url": "https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml"
                },
                {
                    "tags": [
                        "vendor-advisory"
                    ],
                    "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/NRUHDUS2IV2USIZM2CVMSFL6SCKU3RZA/"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/22adf29da8d99933ffed8647d3e0726edd16f7f8"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/40d75c2b7f5c67e254d0a025e0f2e2c7ada7f69f"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/895f7e2ac23eff4743143beef0f0c5ac71ea27d3"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/ba431579efdcbaed7a96f2ac4ea0775879a332fb"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/c62c9e518b784fe44432a3f4fc265fb95b651906"
                },
                {
                    "tags": [
                        "patch"
                    ],
                    "url": "https://github.com/python/cpython/commit/f86b17ac511e68192ba71f27e752321a3252cee3"
                },
                {
                    "url": "http://www.openwall.com/lists/oss-security/2024/06/17/3"
                }
            ],
            "source": {
                "discovery": "UNKNOWN"
            },
            "title": "Incorrect IPv4 and IPv6 private ranges",
            "x_generator": {
                "engine": "Vulnogram 0.2.0"
            }
        },
        "adp": [
            {
                "providerMetadata": {
                    "orgId": "96f1cf3c-f80a-4e0f-a063-99a91e3f1325",
                    "shortName": "secretariat-reference",
                    "dateUpdated": "2024-07-05T18:17:37.383Z"
                },
                "references": [
                    {
                        "tags": [
                            "issue-tracking",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/issues/113171/DEMO_2024_06_19"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/pull/113179/DEMO_2024_06_19"
                    },
                    {
                        "url": "https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml/DEMO_2024_06_19",
                        "tags": [
                            "x_cve_prog_secretariat_snapshot"
                        ]
                    },
                    {
                        "url": "https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml/DEMO_2024_06_19",
                        "tags": [
                            "x_cve_prog_secretariat_snapshot"
                        ]
                    },
                    {
                        "tags": [
                            "vendor-advisory",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/NRUHDUS2IV2USIZM2CVMSFL6SCKU3RZA//DEMO_2024_06_19"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/22adf29da8d99933ffed8647d3e0726edd16f7f8/DEMO_2024_06_19"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/40d75c2b7f5c67e254d0a025e0f2e2c7ada7f69f/DEMO_2024_06_19"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/895f7e2ac23eff4743143beef0f0c5ac71ea27d3/DEMO_2024_06_19"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/ba431579efdcbaed7a96f2ac4ea0775879a332fb/DEMO_2024_06_19"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/c62c9e518b784fe44432a3f4fc265fb95b651906/DEMO_2024_06_19"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_cve_prog_secretariat_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/f86b17ac511e68192ba71f27e752321a3252cee3/DEMO_2024_06_19"
                    },
                    {
                        "url": "http://www.openwall.com/lists/oss-security/2024/06/17/3/DEMO_2024_06_19",
                        "tags": [
                            "x_cve_prog_secretariat_snapshot"
                        ]
                    },
                    {
                        "tags": [
                            "issue-tracking",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/issues/113171/adp-test-6-28"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/pull/113179/adp-test-6-28"
                    },
                    {
                        "url": "https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml/adp-test-6-28",
                        "tags": [
                            "x_snapshot"
                        ]
                    },
                    {
                        "url": "https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml/adp-test-6-28",
                        "tags": [
                            "x_snapshot"
                        ]
                    },
                    {
                        "tags": [
                            "vendor-advisory",
                            "x_snapshot"
                        ],
                        "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/NRUHDUS2IV2USIZM2CVMSFL6SCKU3RZA//adp-test-6-28"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/22adf29da8d99933ffed8647d3e0726edd16f7f8/adp-test-6-28"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/40d75c2b7f5c67e254d0a025e0f2e2c7ada7f69f/adp-test-6-28"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/895f7e2ac23eff4743143beef0f0c5ac71ea27d3/adp-test-6-28"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/ba431579efdcbaed7a96f2ac4ea0775879a332fb/adp-test-6-28"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/c62c9e518b784fe44432a3f4fc265fb95b651906/adp-test-6-28"
                    },
                    {
                        "tags": [
                            "patch",
                            "x_snapshot"
                        ],
                        "url": "https://github.com/python/cpython/commit/f86b17ac511e68192ba71f27e752321a3252cee3/adp-test-6-28"
                    },
                    {
                        "url": "http://www.openwall.com/lists/oss-security/2024/06/17/3/adp-test-6-28",
                        "tags": [
                            "x_snapshot"
                        ]
                    }
                ]
            }
        ]
    },
    "dataVersion": "5.1"
}